Source/WebKit/UIProcess/Cocoa/SOAuthorization/RedirectSOAuthorizationSession.mm

```diff
@@ -67,13 +67,15 @@
     auto* navigationAction = this->navigationAction();
     ASSERT(navigationAction);
     auto* page = this->page();
-    if ((response.httpStatusCode() != 302 && response.httpStatusCode() != 200) || !page) {
+    // FIXME: Enable the useRedirectionForCurrentNavigation code path for all redirections.
```

Comment:
This looks a little strange.  Are we planning to do this in two steps?

Reply:
Yes, we should do it for all 30X redirections, no matter GET or POST.
This will allow removing some iOS specific code for instance.
I limit the changes to 307 POST for now to fix the most urgent issue.